src/index.tsx

```diff
@@ -181,63 +181,70 @@ export function useEffectReducer<
-    }
-
-    const exec = (
-      effect: TEffect | EffectFunction<TState, TEvent, TEffect>
-    ) => {
-      const effectObject = toEffectObject(effect, effectsMap);
-      const effectEntity = createEffectEntity<TState, TEvent, TEffect>(
-        effectObject
-      );
-      nextEffectEntities.push(effectEntity);
+  const wrappedReducer = useMemo(
+    () => (
+      [state, stateEffectTuples, entitiesToStop]: AggregatedEffectsState<
+        TState,
+        TEvent
+      >,
+      event: TEvent | FlushEvent
+    ): AggregatedEffectsState<TState, TEvent> => {
+      const nextEffectEntities: Array<EffectEntity<TState, TEvent>> = [];
+      const nextEntitiesToStop: Array<EffectEntity<TState, TEvent>> = [];
+
+      if (event.type === flushEffectsSymbol) {
+        // Record that effects have already been executed
+        return [
+          state,
+          stateEffectTuples.slice(event.count),
+          nextEntitiesToStop,
+        ];
+      }
 
-      return effectEntity;
-    };
+      const exec = (
+        effect: TEffect | EffectFunction<TState, TEvent, TEffect>
+      ) => {
+        const effectObject = toEffectObject(effect, effectsMap);
+        const effectEntity = createEffectEntity<TState, TEvent, TEffect>(
+          effectObject
+        );
+        nextEffectEntities.push(effectEntity);
 
-    exec.stop = (entity: EffectEntity<TState, TEvent>) => {
-      nextEntitiesToStop.push(entity);
-    };
+        return effectEntity;
+      };
 
-    exec.replace = (
-      entity: EffectEntity<TState, TEvent>,
-      effect: TEffect | EffectFunction<TState, TEvent, TEffect>
-    ) => {
-      if (entity) {
+      exec.stop = (entity: EffectEntity<TState, TEvent>) => {
         nextEntitiesToStop.push(entity);
-      }
-      return exec(effect);
-    };
+      };
+
+      exec.replace = (
+        entity: EffectEntity<TState, TEvent>,
+        effect: TEffect | EffectFunction<TState, TEvent, TEffect>
+      ) => {
+        if (entity) {
+          nextEntitiesToStop.push(entity);
+        }
+        return exec(effect);
+      };
 
-    const nextState = effectReducer(
-      state,
-      event,
-      exec as EffectReducerExec<TState, TEvent, TEffect>
-    );
-
-    return [
-      nextState,
-      nextEffectEntities.length
-        ? [...stateEffectTuples, [nextState, nextEffectEntities]]
-        : stateEffectTuples,
-      entitiesToStop.length
-        ? [...entitiesToStop, ...nextEntitiesToStop]
-        : nextEntitiesToStop,
-    ];
-  };
+      const nextState = effectReducer(
+        state,
+        event,
+        exec as EffectReducerExec<TState, TEvent, TEffect>
+      );
+
+      return [
+        nextState,
+        nextEffectEntities.length
+          ? [...stateEffectTuples, [nextState, nextEffectEntities]]
+          : stateEffectTuples,
+        entitiesToStop.length
+          ? [...entitiesToStop, ...nextEntitiesToStop]
+          : nextEntitiesToStop,
+      ];
+    },
+    []
```

Comment:
This should include `effectReducer` as a dependency since the `wrappedReducer` should be re-created if a new reducer is passed in:

```suggestion
    [effectReducer]
```